k9mail-library/src/test/java/com/fsck/k9/mail/filter/FixedLengthInputStreamTest.java

```diff
@@ -0,0 +1,230 @@
+                "DQzbFjzUJDVlfKNYWISOxPiAIrRvweUFbMRYHJvjskZwemcdoKBiCBwIqOpYrVwrEthvdXgijCtONcTlicKGtYORRhZEJiUIPTSY" +
+                "MEzQrUETaHqkknZFwBKQJJwgZaceaXFmEBLKBoCzcTXayTIEOlGngzfzvkSrDmoOcZroyQSyukxxbJdlaEzDEDJOaSIszGkHeomY" +
+                "KipWsMrKHIYpXaqvstsBlaFRhHyvuwtCdYsjWBtVPDNwTtOuPYklKIHAkvoDceSqonrwWtORQrcZKpiNKzFLqbWgfmnhCcKCeXng" +
+                "uexijkkVShZXIvcYgZdljYyIOQCrlLtHcGMZNJFhCcWJqcriZLRZctrrESxxKZWkGeVcKLFMTHcMwidbiAchSpjQBvCwMXpAlpSt" +
+                "seWUSbdfGQqdadWczKVrEYaiDrhNeepPFDllynREhXUBspdOsRPkVMVaNbcmJEwQkoMeWlsfvRpdAuBZYTzfKVDKQGDoDVDhJyeA" +
+                "BnzjGEEoiGrOiLGoViHpiLCBOLBiLCtfxSWKYuKuixEAJOBiMZFbHGBVDJmUNoeXfxXKNabolkDTCwcPohDrsvvinNCHDSOYnfXp" +
+                "olFRYTClufMizjJMzgBYApgrhOFZpAmmJaTadOgIVVafWvEFKOefUILvQNLaxdWuDVmzOxsSIZiWvYWbpfpIqEMgmfaUpHVdxHOu" +
+                "xbYPqezLbQPjXwULNznEyHYpyCpAFOawJpMmHfluqsSNMzNVlyafSPLDywjvEMIqcYWSmDdcfUQMHbzkECgkrYJuinAWYUPjoJRZ" +
+                "QkxcfkTlhhoFnmbZyvLSrIVcNYdLIDUsqCLQAXEJIJVegUJPfzssOmWHDAPZyLtylUUgFyxvwxLFgAnehydWWXnXIVmnqZEeXeAh" +
+                "cANnXpHqMcAtNXgmBkELyqrYemyPsnlvWMNFJsjJcYNoaAwFiAKVxebtwLLvodQFtuJxasxTcQvsiwSRjWanjWnMDoJKoQtwOcCK" +
+                "mKxHxyUuggufOjgJGvUoUooMBpMXDGrNohNzJbQmnCuFVHxulXEircCYtezkVeEkJKFRDXUlwvOSrfhTmnoytjeQbBcIaRZYsRAn" +
+                "isNlMSHqEXQIDirVGicrcOUxQzYhZlnHqwbKslxGkAjIPesZeQKSgbDMnFaAmVLOOrONYuszgMCNCvNGYJFOWGpZUmNvWvTWFoEX" +
+                "QrPSvEmhxyYmFuIfJppvJSDlggOwcXYiXdtgoFfRZyTQbQmdnhoVHvQffkcMSqHDGRpwSLRbqDWwHJOkVJSMFzHdttJwyVpWuTDa" +
+                "uFXOuHfpoXkHStuKkreyWLsasXTRVchqRaEYjUmcTuNZYlcVPFezKqIaDiRcRxuuYFtMzJqOIwMNmITofhdQiXTIfUHGfsYnPYgM" +
+                "wpVdzGoyIrSSKJjWMnGdhmJHwHoFCqNtXOnbltOMnlCuzBsNUIlrKgfHRTAzAoFAlMvxgIrGffqCLtpAdbCbxzxMuivOctNDhuXy" +
+                "iTuSYUvHnxZxklJDuMbMeGqQddtmAMVAhGNPVfdBaNQiDNxrRqLqFuYMjkDwekzDSgkFbCEiHPwmAqAuoRDrtnwXQfkhyKLNlhRE" +
+                "ZxDybKsCrrQPcrNFKskrvceHPGjvyBtVgaHTSmtzdMSVwNyMIlmsMIgZqpVCNVBmwPVYkEfuenVdGeQcSdGSEMuJGOdWXyMbVQiN" +
+                "hKKwnoffMoKJEbfgSLOMsSVUwEBRGFBPFhhsKLZrmPIdSMKwxhPOvzGywVCamswfIspSUHXyvtQrWGNUkyVyYWwjwumRSVfDFepT" +
+                "JnNHLfbAdkCfHsEbUQCHBlSlgpYExOjYyuYbLjCjBYdNXcfdShRRGhzcdsxmsbAnHunqzIeVsOrlJRfdhaDYtmeIOUBCqFZBpLXT" +
+                "jkvAhXYTjLCmcKebgCUFWCLgYnfxYIrTnbLuEyNPeIlyNIWefQrpYKHtEaUVESWrHYFWGHObaKnpspDilNhkfrWdFXUyZEgkbBPC" +
+                "DSUAsUKPpFNyrWPZCPHzVARrspCJbXNCkotJGtgrDdLMMGIPLEPirRZzRxYgDiefAsUvCTnQXGHTOXTCRgjZCegPBxXTblJVFepV" +
+                "YzPbwggNJDRgrHbtLvEQilSMnHSmeJahMUjJXXHFwZXQlMRVrPolpogKooCqIYErEiNnaWrbOaQhLRVxGCGHVnSaHyqaFQwdaSUF" +
+                "JRjJWyDvRjwIAVYTEghKdtdHnATaBxrExrpBmIdbZciYLtZszlxmziScLwdJWaXVycSAdubKkZGlozHnjBVRqllIzFufoSLgxExE";
+
+        STR_LENGTH = string.length();
+    }
+
+    // Test the cases that fixed length is smaller than the length of bytes array
```

Comment:
We try to avoid adding comments and aim to convey the information using method names or variable names instead.